libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/logger/SalesforceLogger.java

```diff
@@ -0,0 +1,293 @@
+        this.logLevel = level;
+    }
+
+    /**
+     * Disables file logging.
+     */
+    public synchronized void disableFileLogging() {
+        if (fileLogger != null) {
+            fileLogger.setMaxSize(0);
+        }
+    }
+
+    /**
+     * Enables file logging.
+     *
+     * @param maxSize Maximum number of log lines allowed to be stored at a time.
+     */
+    public synchronized void enableFileLogging(int maxSize) {
+        if (fileLogger != null) {
+            fileLogger.setMaxSize(maxSize);
+        }
+    }
+
+    /**
+     * Returns if file logging is enabled or not.
+     *
+     * @return True - if enabled, False - otherwise.
+     */
+    public boolean isFileLoggingEnabled() {
+        int maxSize = 0;
+        if (fileLogger != null) {
+            maxSize = fileLogger.getMaxSize();
+        }
+        return (maxSize > 0);
+    }
+
+    /**
+     * Logs the message passed in at the level specified.
+     *
+     * @param level Log level.
+     * @param tag Log tag.
+     * @param message Log message.
+     */
+    public void log(Level level, String tag, String message) {
+        switch (level) {
+            case OFF:
+                break;
+            case ERROR:
+                Log.e(tag, message);
+                break;
+            case WARN:
+                Log.w(tag, message);
+                break;
+            case INFO:
+                Log.i(tag, message);
+                break;
+            case DEBUG:
+                Log.d(tag, message);
+                break;
+            case VERBOSE:
+                Log.v(tag, message);
+                break;
+            default:
+                Log.d(tag, message);
+        }
+        if (level != Level.OFF) {
+            logToFile(level, tag, message, null);
+        }
+    }
+
+    /**
+     * Logs the message and throwable passed in at the level specified.
+     *
+     * @param level Log level.
+     * @param tag Log tag.
+     * @param message Log message.
+     * @param e Throwable to be logged.
+     */
+    public void log(Level level, String tag, String message, Throwable e) {
+        switch (level) {
+            case OFF:
+                break;
+            case ERROR:
+                Log.e(tag, message, e);
+                break;
+            case WARN:
+                Log.w(tag, message, e);
+                break;
+            case INFO:
+                Log.i(tag, message, e);
+                break;
+            case DEBUG:
+                Log.d(tag, message, e);
+                break;
+            case VERBOSE:
+                Log.v(tag, message, e);
+                break;
+            default:
+                Log.d(tag, message, e);
+        }
+        if (level != Level.OFF) {
+            logToFile(level, tag, message, e);
+        }
+    }
+
+    private void logToFile(final Level level, final String tag, final String message, final Throwable e) {
+        THREAD_POOL.execute(new Runnable() {
```

Comment:
I don't think so. The underlying file write API in `Tape` is `synchronized`, so order is guaranteed.